File: python/leetcode/String-Manipulation/group-anagrams.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import collections

def groupAnagrams(strs):
	anagrams = collections.defaultdict(list)

	for word in strs:
		anagrams[''.join(sorted(word))].append(word)
	return list(anagrams.values())

strs = ["eat","tea","tan","ate","nat","bat"]
print(groupAnagrams(strs))

File: js/leetcode/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
const groupAnagrams = function (strs) {
  const anangrams = {};

  strs.forEach((str) => {
    const key = str.split('').sort().join('');
    if (!anangrams[key]) anangrams[key] = [str];
    else anangrams[key].push(str);
  });
  return Object.values(anangrams);
};

console.log(groupAnagrams(['eat', 'tea', 'tan', 'ate', 'nat', 'bat']));
console.log(groupAnagrams(['']));
console.log(groupAnagrams(['a']));
